server_cmds.go

```diff
@@ -351,19 +365,66 @@ func (s *ServerMethods) PrivilegeKeyAdd(ttype, id1, id2 int, options ...CmdArg)
 
 // OnlineClient represents a client online on a virtual server.
 type OnlineClient struct {
-	ID          int    `ms:"clid"`
-	ChannelID   int    `ms:"cid"`
-	DatabaseID  int    `ms:"client_database_id"`
-	Nickname    string `ms:"client_nickname"`
-	Type        int    `ms:"client_type"`
+	// Following variables are always returned by ClientList().
+	ID         int    `ms:"clid"`
+	ChannelID  int    `ms:"cid"`
+	DatabaseID int    `ms:"client_database_id"`
+	Nickname   string `ms:"client_nickname"`
+	Type       int    `ms:"client_type"`
+	// Following variables are optional and can be requested in ClientList() to get extended client information.
+	// note: Away and AwayMessage are currently optional but not using pointers for compatibility with older versions of teamspeakserver.
+	// This means they will always be returned, but their value will only be representative when the ClientAway option is specified.
+	OnlineClientAway   `ms:",squash"` // ts3.ClientAway
+	UniqueIdentifier   *string        `ms:"client_unique_identifier"` // ts3.ClientUid
+	OnlineClientVoice  `ms:",squash"` // ts3.ClientVoice
+	OnlineClientTimes  `ms:",squash"` // ts3.ClientTimes
+	OnlineClientGroups `ms:",squash"` // ts3.ClientGroups
+	OnlineClientInfo   `ms:",squash"` // ts3.ClientInfo
```

Comment:
We can make the embeds pointers to avoid duplicate decoding checks and pointer dereference.

While not ideal as we're at v1.xx we should fully revert the away changes to ensure full compatibility.

I'm assuming with `OnlineClientAway` won't be populated without `ClientAway` option? If so we should comment to make that clear.

Also as we're in the package we shouldn't include ts3 in the comment.

Note: suggestion may need reformatting
```suggestion
       	Away        bool   `ms:"client_away"` // Only populated if ClientAway or ClientListFull is passed to ClientList.
	AwayMessage string `ms:"client_away_message"` // Only populated if ClientAway or ClientListFull is passed to ClientList.
	UniqueIdentifier   *string        `ms:"client_unique_identifier"` // Only populated if ClientUid or ClientListFull is passed to ClientList.
	*OnlineClientVoice  `ms:",squash"` // Only populated if ClientVoice or ClientListFull is passed to ClientList. 
	*OnlineClientTimes  `ms:",squash"` // Only populated if ClientTimes or ClientListFull is passed to ClientList. 
	*OnlineClientGroups `ms:",squash"` // Only populated if ClientGroups or ClientListFull is passed to ClientList. 
	*OnlineClientInfo   `ms:",squash"` // Only populated if ClientInfo or ClientListFull is passed to ClientList.
```

Reply:
> We can make the embeds pointers to avoid duplicate decoding checks and pointer dereference.

I have tried implementing this in the newest commit.

> While not ideal as we're at v1.xx we should fully revert the away changes to ensure full compatibility.

Ok, I understand. I should have now reverted this to how it used to be originally.

> I'm assuming with OnlineClientAway won't be populated without ClientAway option? If so we should comment to make 
that clear.

Agreed, added in comment in new commit.

> Also as we're in the package we shouldn't include ts3 in the comment.

I have taken your example and added the new comments in the new commit.